azure/monitoring.tf

```diff
@@ -197,6 +198,7 @@ type: Opaque
 }
 
 resource "kubectl_manifest" "thanos-datasource" {
+  count      = var.thanos_enabled ? 1 : 0
   depends_on = [kubectl_manifest.thanos-datasource-credentials]
```

Comment:
I think you need to update all of your references with `[0]` for the resources that you are now optionally enabling.